test/extensions/filters/http/file_system_buffer/filter_integration_test.cc

```diff
@@ -0,0 +1,176 @@
+#include "test/integration/http_protocol_integration.h"
+
+namespace Envoy {
+namespace {
+
+absl::string_view testTmpDir() {
+  auto env_tmpdir = std::getenv("TEST_TMPDIR");
+  if (env_tmpdir) {
+    return env_tmpdir;
+  }
+  env_tmpdir = std::getenv("TMPDIR");
+  return env_tmpdir ? env_tmpdir : "/tmp";
+}
+
+// A config which will trigger buffering and injecting content length into headers.
+std::string contentLengthConfig() {
+  return absl::StrCat(
+      R"EOF(
+        name: file_system_buffer
+        typed_config:
+          "@type": type.googleapis.com/envoy.extensions.filters.http.file_system_buffer.v3.FileSystemBufferFilterConfig
+          manager_config:
+            thread_pool:
+              thread_count: 1
+          request:
+            memory_buffer_bytes_limit: 4096
+            storage_buffer_bytes_limit: 8000000
+            behavior:
+              fully_buffer_and_always_inject_content_length: {}
+          response:
+            memory_buffer_bytes_limit: 4096
+            storage_buffer_bytes_limit: 8000000
+            behavior:
+              fully_buffer_and_always_inject_content_length: {}
+          storage_buffer_path: )EOF",
+      testTmpDir());
+}
+
+// A config with a small request buffer, to test the behavior of exceeding the limit.
+std::string smallRequestBufferConfig() {
+  return absl::StrCat(
+      R"EOF(
+        name: file_system_buffer
+        typed_config:
+          "@type": type.googleapis.com/envoy.extensions.filters.http.file_system_buffer.v3.FileSystemBufferFilterConfig
+          manager_config:
+            thread_pool:
+              thread_count: 1
+          request:
+            memory_buffer_bytes_limit: 64
+            storage_buffer_bytes_limit: 64
+            behavior:
+              fully_buffer: {}
+          storage_buffer_path: )EOF",
+      testTmpDir());
+}
+
+// Integration test //
+
+using FileSystemBufferIntegrationTest = HttpProtocolIntegrationTest;
```

Comment:
Please run these new tests with runs_per_test using the deflake instructions to check for flakes.https://github.com/envoyproxy/envoy/tree/main/test/integration#reproducing-test-flakes